com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ExtenderUtil.java

```diff
@@ -416,6 +409,40 @@ public static void mergeBundleMap(Map<String, IResource> into, Map<String, IReso
         }
     }
 
+    /**
+     * Given a project path to ext.manifest, return an extension root dir if this path has engine extensions
```

Comment:
I think either the comment is wrong, of the function name is wrong.

"Given a project path to ext.manifest, return an extension root dir"
The directory of the `ext.manifest` _is_ the extension root dir.
So if one has found the manifest, one doesn't really need to "find" the root dir.

I suggest renaming it to "hasNativeExtensionSource(String path)" (I don't like the "maybe" part of the name)

Reply:
> The directory of the ext.manifest _is_ the extension root dir.

Yes, and there is no contradiction between this statement and method documentation. It returns the parent dir but does extra checks to ensure the dir has sources. The `maybe` part is to signify that it might return `null`. Do you think it's better to instead return `Optional` to denote that not every dir with `ext.manifest` has engine extensions?

Reply:
`hasNativeExtensionSource` implies a boolean, but the method returns a path or null, not sure it's better...

Reply:
I also wanted to return a string or null instead of a boolean to avoid duplicating the `f.getParent() == null ? "" : f.getParent()` logic.

Reply:
True, perhaps "get" is better.
As long as we avoid "maybe" :)